### PR TITLE
Fix completed card button

### LINKED ISSUE #33211: 'Show setup task list' button is not working.
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

After finishing the setup tasklist it's not possible to see the setup task list again without refreshing the page. The button `Show setup task list` in the kebab menu is not working.

screenshot-glamorous-echidna jurassic ninja-2022 05 26-13_33_36

### Expected behavior

After pressing `Show setup task list` the setup tasklist should be visible again. 

### Actual behavior

The button `Show setup task list` is not working.

Screen Capture on 2022-05-26 at 13-19-57

### Steps to reproduce

1. Start with a fresh install
2. Install and activate the latest [WooCommerce Test Helper plugin](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/tag/v0.7.4)
3. Navigate to **Tools > WCA Test Helper > Experiments** and toggle `woocommerce_tasklist_setup_experiment_2_2022_*` to treatment.
4. Finish all the tasks.
5. The message `You've completed store setup` will be visible.
6. Select `Show setup task list` in the kebab menu. 
7. Nothing happens.

### WordPress Environment

WordPress 6.0
WooCommerce 6.6.0

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR fixes the button `Show setup task list` in the completed setup card. 

Closes #33211.

### How to test the changes in this Pull Request:

1. Start with a fresh install
2. Install and activate the latest [WooCommerce Test Helper plugin](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/tag/v0.7.4)
3. Navigate to **Tools > WCA Test Helper > Experiments** and toggle `woocommerce_tasklist_setup_experiment_2_2022_*` to treatment.
4. Finish all the tasks.
5. The message `You've completed store setup` will be visible.
6. Select `Show setup task list` in the kebab menu. 
7. The setup task list should be visible now.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33238/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Update sectioned task list to use onboarding data store` 16af93b2d945fb9c2ec18a227364cc6cedfb97db
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx [hunk 1]
```diff
@@ -54,6 +54,10 @@ export const SectionedTaskList: React.FC< TaskListProps > = ( {
 	const [ openPanel, setOpenPanel ] = useState< string | null >(
 		sections?.find( ( section ) => ! section.isComplete )?.id || null
 	);
+	const [
+		isCompletedCardVisible,
+		setIsCompletedCardVisible,
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx]: Took me a minute to grok what was happening, but I think a more intentional name might be `shouldKeepTasks`, what do you think?

File: plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx [hunk 2]
```diff
@@ -117,7 +122,11 @@ export const SectionedTaskList: React.FC< TaskListProps > = ( {
 		return <div className="woocommerce-task-dashboard__container"></div>;
 	}
 
-	if ( isComplete && keepCompletedTaskList !== 'yes' ) {
+	if (
+		isComplete &&
+		keepCompletedTaskList !== 'yes' &&
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx]: I assume just relying on this causes a delay and we want to optimistically show the task list again before this option is re-retrieved, right?

This is another use case for our generic dismissal API.  Would love to discuss getting this into an upcoming cycle.  cc @mattsherman

File: plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce-admin/client/two-column-tasks/sectioned-task-list.tsx]: Good idea! I renamed the const in commit [f621b4e](https://github.com/woocommerce/woocommerce/pull/33238/commits/f621b4e22e3031e2ae9e4ed5eb4f6d2b61d5c82b)

**Review by louwie17 — APPROVED** [review]
Thanks for working on this @octaedro and @joshuatf, this tested well and changes look good. 🎉

**Comment by jonathansadowski** [comment]
This has been cherry-picked into the `release/6.6` branch.

**Pull request merged**